include/boost/math/tools/polynomial.hpp

```diff
@@ -558,59 +562,52 @@ class polynomial :
 
 
 template <class T>
-inline polynomial<T> operator + (const polynomial<T>& a, const polynomial<T>& b)
+inline polynomial<T> operator + (polynomial<T> a, const polynomial<T>& b)
 {
-   polynomial<T> result(a);
-   result += b;
-   return result;
+   a += b;
+   return a;
```

Comment:
Sorry about the messed up formatting, haven't figured this stupid interface out yet :(